src/Components/Main.tsx

```diff
@@ -86,145 +88,200 @@ export const Main = (props: IProps) => {
     return comparison;
   };
 
-  const filteredRequests = requests.slice(0, props.maxRequests).filter((request) => {
-    if (filter === EnumFilterType.All) {
-      if (request instanceof ReduxAction) {
-        return request.action.type.toLowerCase().includes(searchQuery.toLowerCase());
+  const filteredRequests = (): Array<any> =>
+    requests.slice(0, props.maxRequests).filter((request) => {
+      if (filter === EnumFilterType.All) {
+        if (request instanceof ReduxAction) {
+          return JSON.stringify(request.action).toLowerCase().includes(searchQuery.toLowerCase());
+        }
       }
-    }
-    // TODO: Add  || request instanceof NRequest when Native request will be available
-    if (request instanceof RNRequest) {
-      if (filter === request.method || filter === EnumFilterType.All) {
-        return (
-          request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
-          request.status === parseInt(searchQuery) ||
-          request.method.toLowerCase() === searchQuery.toLowerCase()
-        );
+      if (request instanceof RNRequest || request instanceof NRequest) {
+        if (request.type === EnumSourceType.ReactNativeRequest) {
+          if (filter === request.method || filter === EnumFilterType.All) {
+            return (
+              request.type === EnumSourceType.ReactNativeRequest &&
+              (request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
+                request.status === parseInt(searchQuery, 10) ||
+                request.method.toLowerCase() === searchQuery.toLowerCase())
+            );
+          }
+        } else {
+          if (filter === request.method || filter === EnumFilterType.All) {
+            return (
+              request.type === EnumSourceType.Nativerequest &&
+              (request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
+                request.status === parseInt(searchQuery, 10) ||
+                request.method.toLowerCase() === searchQuery.toLowerCase())
+            );
+          }
+        }
       }
-    }
-    return;
-  });
+      return false;
+    });
 
   const clearList = () => {
     props.clearAll();
     setRequests([]);
   };
 
+  const formatSharedMessage = (array: Array<ILog>): string => {
```

Comment:
should be moved into utils/helpers

Reply:
You are right, can I move this part in my current branche? (feat/share)